### ISSUE TITLE
404 when wildcard value starts with "p"

### ISSUE THREAD

**Issue by kabukky** [dimfeld/httptreemux#8]
Hey,
I'm experiencing a very peculiar issue.
Let's say my server is at http://127.0.0.1:8084 and I'm using this setup:

    mux := httptreemux.New()
    mux.GET("/:slug/", postHandler)

postHandler get's called every time (well, every time I've tried so far) EXCEPT when I'm opening:

    http://127.0.0.1:8084/patch/
    http://127.0.0.1:8084/post-test/
    http://127.0.0.1:8084/prmmmmm/

and (so far) every other url with a slug that starts with "p". I haven't tried all of the other letters yet.

A few things to note:
- This doesn't happen with `mux.GET("/tag/:slug/", tagHandler)` and a slug that starts with "p". So it might only happen when the wildcard is the first part of the path.
- When I open e. g. `http://127.0.0.1:8084/patch` it doesn't get redirected to `http://127.0.0.1:8084/patch/`, so the bug might be before the redirection happens?

Thanks

/EDIT: I found the same issue with "a". all other lower case letters are fine. I haven't tried any special characters except for öäü.

**Comment by dimfeld** [comment]
I'm not seeing this. I tried the test code below and it works fine. Can you post your entire route map? It sounds like there may be some bad interaction there between the route you showed and some others.

```
func TestRootWildcard(t *testing.T) {
	var suppliedParam string
	simpleHandler := func(w http.ResponseWriter, r *http.Request, params map[string]string) {
		suppliedParam = params["slug"]
	}

	router := New()
	router.GET("/:slug", simpleHandler)

	r := newRequest("GET", "/patch", nil)
	w := httptest.NewRecorder()
	router.ServeHTTP(w, r)

	if suppliedParam != "patch" {
		t.Errorf("Expected param patch, saw %s", suppliedParam)
	}

	if w.Code != http.StatusOK {
		t.Errorf("Expected status 200, saw %d", w.Code)
	}
}
```

**Comment by dimfeld** [comment]
Ok, this route map reproduces the failure:
```
router.GET("/:page")
router.GET("/paper")
router.GET("/post/:post")
```

I'll get a fix out soon.

**Comment by kabukky** [comment]
Hi dimfeld,
that sounds about right for me. This is my route map.
`blog.go`:

```
	// For index
	router.GET("/", indexHandler)
	router.GET("/:slug/", postHandler)
	router.GET("/page/:number/", indexHandler)
	// For author
	router.GET("/author/:slug/", authorHandler)
	router.GET("/author/:slug/:function/", authorHandler)
	router.GET("/author/:slug/:function/:number/", authorHandler)
	// For tag
	router.GET("/tag/:slug/", tagHandler)
	router.GET("/tag/:slug/:function/", tagHandler)
	router.GET("/tag/:slug/:function/:number/", tagHandler)
	// For serving asset files
	router.GET("/assets/*filepath", assetsHandler)
	router.GET("/images/*filepath", imagesHandler)
	router.GET("/public/*filepath", publicHandler)
```

`admin.go`:
```
	// For admin panel
	router.GET("/admin/", adminHandler)
	router.GET("/admin/login/", getLoginHandler)
	router.POST("/admin/login/", postLoginHandler)
	router.GET("/admin/register/", getRegistrationHandler)
	router.POST("/admin/register/", postRegistrationHandler)
	router.GET("/admin/logout/", logoutHandler)
	router.GET("/admin/*filepath", adminFileHandler)

	// For admin API (no trailing slash)
	// Posts
	router.GET("/admin/api/posts/:number", apiPostsHandler)
	// Post
	router.GET("/admin/api/post/:id", getApiPostHandler)
	router.POST("/admin/api/post", postApiPostHandler)
	router.PATCH("/admin/api/post", patchApiPostHandler)
	router.DELETE("/admin/api/post/:id", deleteApiPostHandler)
	// Upload
	router.POST("/admin/api/upload", apiUploadHandler)
	// Images
	router.GET("/admin/api/images/:number", apiImagesHandler)
	router.DELETE("/admin/api/image", deleteApiImageHandler)
	// Blog
	router.GET("/admin/api/blog", getApiBlogHandler)
	router.PATCH("/admin/api/blog", patchApiBlogHandler)
	// User
	router.GET("/admin/api/user/:id", getApiUserHandler)
	router.PATCH("/admin/api/user", patchApiUserHandler)
	// User id
	router.GET("/admin/api/userid", getApiUserIdHandler)
```

**Comment by dimfeld** [comment]
Fixed in 540528aebaa892c32e0e1dc451345d0fe9317b82. Thanks for the report! Should get the fix for #7 up this weekend as well.

**Comment by kabukky** [comment]
Works like a charm. Thank you :)
